services/controllerhost/api_handlers.go

```diff
@@ -353,6 +357,13 @@ func refreshInputHostsForDst(context *Context, dstUUID string, now int64) ([]str
 	}
 
 	var dstType = getDstType(dstDesc)
+
+	// Fail attempts to publish to Kafka destinations
+	if dstType == dstTypeKafka {
```

Comment:
how about also storing the an empty set of ip:ports as result in the result cache ? That way, if someone mistakenly tries to publish to a kafka destination and calls getInputHosts() in a loop, we won't do a metadata.readDestination() for every call.